src/common/factory.ts

```diff
@@ -48,4 +66,73 @@ export class TurboBaseFactory {
       paymentService,
     });
   }
+
+  static authenticated({
+    privateKey,
+    signer: providedSigner,
+    paymentServiceConfig = {},
+    uploadServiceConfig = {},
+    token = 'arweave',
+    gatewayUrl,
+    tokenTools,
+  }: TurboAuthenticatedConfiguration) {
+    let signer: TurboSigner;
+
+    if (providedSigner) {
+      signer = providedSigner;
+
+      // Derive token from signer if not provided
+      if (!token) {
+        if (signer instanceof EthereumSigner) {
+          token = 'ethereum';
+        } else if (signer instanceof HexSolanaSigner) {
+          token = 'solana';
+        }
+      }
+    } else if (privateKey) {
+      if (token === 'solana') {
+        // TODO: test wallet creation
+        signer = new HexSolanaSigner(privateKey);
+      } else {
+        signer = new ArweaveSigner(privateKey);
+      }
+    } else {
+      throw new Error('A privateKey or signer must be provided.');
+    }
+
+    // when in browser, we use TurboWebArweaveSigner
+    const turboSigner =
+      typeof window !== 'undefined'
+        ? new TurboWebArweaveSigner({
+            signer,
+            logger: this.logger,
+          })
+        : new TurboNodeSigner({
+            signer,
+            logger: this.logger,
+          });
```

Comment:
this was previously only done in the web exported version - any reason for moving it up here?

Reply:
the code in these factories seemed completely duplicated. I'll re-review that

Reply:
changed that back... looks like it was causing build errors 👀